src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
-
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= (b + c) || b >= (a + c) || c >= (a + b)) {
+      throw new Error('invalid side length for triangle');
+    }
+
+    this.shape = Shape.Triangle;
```

Comment:
you already did it on lines 16

Reply:
![image](https://github.com/mate-academy/ts_figures_classes/assets/51528776/3c5c3731-dd29-4f85-a31d-ff6627e84ee7)